### PR TITLE
Sort Helm parameters alphabetically in .argocd-source-<appName>.yaml

### PR DESCRIPTION
**Pull request by christian-schlichtherle**
... because this results in deterministic output which plays nicely with diff based tools like `git diff` etc.

## Summary by CodeRabbit

* **Bug Fixes**
  * Helm parameter entries in application overrides are now deterministically sorted by name to prevent spurious diffs while preserving all parameter fields and flags.

* **Documentation**
  * Updated Git write-back docs to note Helm parameters are alphabetically sorted for deterministic output; changelog updated.

* **Tests**
  * Added tests verifying stable Helm parameter ordering across edge cases and preservation of all fields.

### CODE REVIEW

File: pkg/argocd/update.go [hunk 1]
```diff
@@ -507,6 +510,18 @@ func marshalParamsOverride(ctx context.Context, applicationImages *ApplicationIm
 	return override, nil
 }
 
+func sortHelmParameters(params []v1alpha1.HelmParameter) {
+	slices.SortFunc(params, func(a, b v1alpha1.HelmParameter) int {
+		if a.Name < b.Name {
```
Comment (dkarpele) [inline pkg/argocd/update.go]: Please simplify this with `cmp.Compare(a.Name, b.Name)`

File: pkg/argocd/update.go [hunk 2]
```diff
@@ -507,6 +510,18 @@ func marshalParamsOverride(ctx context.Context, applicationImages *ApplicationIm
 	return override, nil
 }
 
+func sortHelmParameters(params []v1alpha1.HelmParameter) {
```
Comment (dkarpele) [inline pkg/argocd/update.go:514]: Please add a separate unit test for `sortHelmParameters` to update_test.go.

**Review by dkarpele — COMMENTED** [review]
Please sign-off your commits

**Comment by christian-schlichtherle** [comment]
I'm done with the requested changes, please review.

**Comment by christian-schlichtherle** [comment]
I've also updated the docs and the CHANGELOG.md - please review.

**Review by chengfang — APPROVED** [review]
